src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use thiserror::Error;
+
+use crate::{
+    parse::{Context, Parse},
+    text::simple_extensions,
+};
+
+/// A parsed [simple_extensions::ArgumentsItem].
+#[derive(Clone, Debug)]
+pub enum ArgumentsItem {
+    /// Arguments that support a fixed set of declared values as constant arguments.
+    EnumArgument(EnumerationArg),
+
+    /// Arguments that refer to a data value.
+    ValueArgument(ValueArg),
+
+    /// Arguments that are used only to inform the evaluation and/or type derivation of the function.
+    TypeArgument(TypeArg),
+}
+
+impl ArgumentsItem {
+    #[inline]
+    fn parse_name(name: Option<String>) -> Result<Option<String>, ArgumentsItemError> {
+        match name {
+            Some(s) if s.is_empty() => {
+                Err(ArgumentsItemError::EmptyOptionalField("name".to_string()))
+            }
+            _ => Ok(name),
+        }
+    }
+
+    #[inline]
+    fn parse_description(
+        description: Option<String>,
+    ) -> Result<Option<String>, ArgumentsItemError> {
+        match description {
+            Some(s) if s.is_empty() => Err(ArgumentsItemError::EmptyOptionalField(
+                "description".to_string(),
+            )),
+            _ => Ok(description),
+        }
+    }
+}
+
+impl<C: Context> Parse<C> for simple_extensions::ArgumentsItem {
+    type Parsed = ArgumentsItem;
+    type Error = ArgumentsItemError;
+
+    fn parse(self, ctx: &mut C) -> Result<Self::Parsed, Self::Error> {
+        match self {
+            simple_extensions::ArgumentsItem::EnumerationArg(arg) => Ok(ctx.parse(arg)?.into()),
+            simple_extensions::ArgumentsItem::ValueArg(arg) => Ok(ctx.parse(arg)?.into()),
+            simple_extensions::ArgumentsItem::TypeArg(arg) => Ok(ctx.parse(arg)?.into()),
+        }
+    }
+}
+
+impl From<ArgumentsItem> for simple_extensions::ArgumentsItem {
+    fn from(value: ArgumentsItem) -> Self {
+        match value {
+            ArgumentsItem::EnumArgument(arg) => arg.into(),
+            ArgumentsItem::ValueArgument(arg) => arg.into(),
+            ArgumentsItem::TypeArgument(arg) => arg.into(),
+        }
+    }
+}
+
+/// Parse errors for [simple_extensions::ArgumentsItem].
+#[derive(Debug, Error, PartialEq)]
+pub enum ArgumentsItemError {
+    /// Invalid enumeration options.
+    #[error("invalid enumeration options: {0}")]
+    InvalidEnumOptions(#[from] EnumOptionsError),
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumerationArg {
```

Comment:
Can you add an impl with methods to return references to the fields of this struct? Similar to for example: https://docs.rs/substrait/latest/substrait/parse/proto/struct.Version.html#impl-Version.

Reply:
Added in [commit](https://github.com/substrait-io/substrait-rs/pull/185/commits/4713c54f3bf70553ac9fdbe389264616fcf910e6).